package.json

```diff
@@ -17,7 +17,8 @@
     "lint:web": "eslint web",
     "i18n": "npm run-script i18n:init && npm run-script i18n:extract",
     "i18n:init": "ngx-translate-extract --input ./web --output ./web/public/assets/i18n/template.json --key-as-default-value --replace --format json",
-    "i18n:extract": "ngx-translate-extract --input ./web --output ./web/public/assets/i18n/en.json --clean --format json"
+    "i18n:extract": "ngx-translate-extract --input ./web --output ./web/public/assets/i18n/en.json --clean --format json",
+    "version": "genversion -e src/packageVersion.ts && git add src/packageVersion.ts"
```

Comment:
this puts a lot of requirement on how dimension is released, which is not great.

What is the actual issue with trying to import the package.json?

Reply:
When trying to import package.json, you're forced to have the package.json at a specific position. This isn't ideal because you're essentially forcing a specific location that the whole project must be. This has multiple issues:

* The compiled version that gets saved in `root/dist/` is in the wrong location as when it tries to import, it looks up at `root/dist/package.json` instead of `root/package.json`.
* Which also means if we decide to just import as something like `../../package.json`, you then break the typescript source as it looks one folder behind root. This would potentially be a security issue if it goes beyond project root.
* And lastly the user must always keep a specific folder structure to have the package.json import work. You can't just take the content of dist and move it to a different location. You must set it so that the relative path of package.json fits at all times.

The final question to what it's solving is: How can you make `../package.json` work when the src has the package at `../` while dist has the package at `../../`? Moving the package.json file would technically work for the Docker image. But for bare metal installation, it would break.

Since package.json is something that is pretty delicate, and you need to have it as the "root" file, you're getting very limited with how you keep the src and the dist build. Having a file get auto-generated by taking the version of package.json seems like a logical way to have the code work at any situation as it imports from the same folder. I assumed that `npm version` is the go-to usage for generating new version, but I wasn't 100% sure as I think there was never a full version release.

Reply:
Something like https://github.com/sindresorhus/read-pkg-up would probably work well, I think. The versioning for projects like this is typically isolated from a simple `npm version` command, and we only need the version in the backend. `genversion` works great if we also have to worry about the client-side, but we don't in this case.

Reply:
I have replayed this with read-pkg-up and removed the version script. The commit mentions it, but I had to go with v7 instead of the latest as newer versions are ESM-only and do not work with the way Dimension is built right now.